src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,169 @@
+     */
+    public void createDatabase() {
+        File database = new File(DATABASE_PATH.toString());
+
+        try {
+            database.getParentFile().mkdirs();
+            database.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Checks if the database exists.
+     *
+     * @return True if the text file exists.
+     */
+    public boolean databaseExist() {
+        File database = new File(DATABASE_PATH.toString());
+        return database.exists();
+    }
+
+
+    /**
+     * Parses the string of tasks from the database into Task objects.
+     *
+     * @return The list of tasks stored in the database.
+     * @throws DukeException If corrupted database.
+     */
+    protected ArrayList<Task> readFromDatabase() throws DukeException {
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        try {
+            FileReader reader = new FileReader(DATABASE_PATH.toString());
+            BufferedReader bufferedReader = new BufferedReader(reader);
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                taskList.add(stringToTask(line));
+            }
+            reader.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        return taskList;
+    }
+
+    /**
+     * Updates the database.
+     *
+     * @param taskList The list containing the tasks to be saved.
+     */
+    public void writeToDatabase(ArrayList<Task> taskList) {
+        try {
+            FileWriter writer = new FileWriter(DATABASE_PATH.toString(), false);
+            BufferedWriter bufferedWriter = new BufferedWriter(writer);
+
+            for (Task task : taskList) {
+                bufferedWriter.write(task.toDatabaseRepresentation());
+                bufferedWriter.newLine();
+            }
+
+            bufferedWriter.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Task stringToTask(String line) throws DukeException {
+        try {
+            Task newTask;
+            String[] taskInfo = line.split(" \\| ");
+
+            switch (taskInfo[0]) {
+                case "T":
+                    newTask = new Todo(taskInfo[2]);
+                    break;
+                case "D":
+                    newTask = new Deadline(taskInfo[2], taskInfo[3]);
+                    break;
+                case "E":
+                    newTask = new Event(taskInfo[2], taskInfo[3], taskInfo[4]);
+                    break;
+                default:
+                    throw new DukeException("Database is corrupted.");
+            }
+
+            if (taskInfo[1].equals("1")) {
+                newTask.markAsDone();
+            }
+
+            return newTask;
+        } catch (DukeException e) {
+            throw e;
+        }
+    }
```

Comment:
Interesting to see that the creation of Task and its subclasses were handled in Storage. Maybe this was part of an old implementation so it may be a good idea to remove it to prevent confusion?